plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -89,7 +92,47 @@ var BasicsChart = React.createClass({
       basicsData = _.assign({}, basicsData, _.cloneDeep(basicsState));
       var dataMunger = dataMungerMkr(this.props.bgClasses);
       dataMunger.reduceByDay(basicsData);
-      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData);
+
+      // get site changes from three event types
+      basicsData.data.cannulaPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_CANNULA);
+      basicsData.data.tubingPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_TUBING);
+      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_RESERVOIR);
+
+      var siteChangeEvents = [];
+
+      if (basicsData.data.tubingPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.tubingPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.cannulaPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.cannulaPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.reservoirChange.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.reservoirChange.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+
+      if (siteChangeEvents.length) {
+        // get latest pump upload:
+        var latestPump = _.last(_.sortBy(siteChangeEvents, 'time')).source;
```

Comment:
Regarding "latest pump" — interesting, that was not my takeaway from the Trello card, but perhaps I don't have the background on what "older pump" and "latest pump" mean? That was one point I was going to clarify with Brandon and Sara. My expectation would be that this section shows "site changes" from the most recent pump event, whether it's reservoir, tubing, or cannula. I see what you're saying, where if you've been using Tandem and then switch to Medtronic, you wouldn't have (nor would you be able to switch to) the Tandem site changes based on cannula or prime, because the Medtronic reservoir changes would override. (An aside: if you switch between pumps, would you need to change a site, anyway? I know Medtronic doesn't use the Luer lock, but I'm not sure if the site-end of their proprietary tubing is interchangeable.) I'll see what they say, and update accordingly.